tests/helper/files.go

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+// Simplifies work with file testing
+package helper
+
+import (
+	"bytes"
+	"crypto/md5" // #nosec G501
+	"crypto/rand"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+// CreateRandomFiles will take directory and put there as much random files as you want
+func CreateRandomFiles(dir string, amount int) ([]string, error) {
```

Comment:
Introducing non-determinism in the tests may result in significant debugging challenges.  It might be advisable to instead have a `CreateFiles` that takes in a map of `filename => file data` pairs.  That would make it so the test needs to explicitly enumerate its expected files, which will likely keep it more accessible long term.

The way its used, `"testfile_"+strconv.Itoa(i)` will result in exact filenames being able to be inferred, but only because the test is "peeking into implementation details" of this method.

Reply:
Random is needed here to make sure any sort of caching is out of question. Random affects just binary data, the file names are quite deterministic. So since we don't care about the binaries within the file (only scp protocol cares about that, which is tested by the other people) it's a good way to keep best of both worlds - other ways are too complicated, so leaving as is until we will face some serious issues.

Reply:
Works for me, yes, the filenames are deterministic and that's what matters.